### PR TITLE
Fix json iterate

### PR DESCRIPTION
**Pull request by jesper-friis**
# Description
Cleaned up iteration over instances in a JSON file.

Added `dlite.Instance.get_uuids()` as a convenient way to list instance UUIDs in a storage.

## Type of change
- [x] Bug fix & code cleanup
- [x] New feature
- [ ] Documentation update
- [ ] Test update

## Checklist for the reviewer
This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [ ] Is the code easy to read and understand?
- [ ] Do all new feature have an accompanying new test?
- [ ] Has the documentation been updated as necessary?

### CODE REVIEW

File: src/utils/jsmnx.c [hunk 1]
```diff
@@ -24,42 +24,46 @@
...
  */
 int jsmn_parse_alloc(jsmn_parser *parser, const char *js, const size_t len,
                      jsmntok_t **tokens_ptr, unsigned int *num_tokens_ptr)
 {
   int n, n_save;
-  unsigned int saved_pos;
   jsmntok_t *t=NULL;
   (void) n_save;  // avoid unused parameter error when assert is turned off
   assert(tokens_ptr);
   assert(num_tokens_ptr);
```
Comment (torhaugl) [inline src/utils/jsmnx.c:39]: Could you rename n to n_tokens, and t to tokens?

**Review by torhaugl — COMMENTED** [review]
I left some comments on the JSON parser to improve readability. The short names makes the code a bit tricky to understand.

File: src/utils/jsmnx.c (hunk 1 above)
Comment (jesper-friis) [reply src/utils/jsmnx.c:39]: done

File: src/utils/jsmnx.c [hunk 2]
```diff
@@ -24,42 +24,46 @@
...
   if (!*num_tokens_ptr) *tokens_ptr = NULL;
   if (!*tokens_ptr) *num_tokens_ptr = 0;
 
-  saved_pos = parser->pos;
-
   if (!*tokens_ptr) {
-    if ((n = jsmn_parse(parser, js, len, NULL, 0)) < 0) goto fail;
+    jsmn_parser tmp_parser;
+    jsmn_init(&tmp_parser);
+    if ((n = jsmn_parse(&tmp_parser, js, len, NULL, 0)) < 0) goto fail;
```
[the remark this answers is not in the document]
Comment (jesper-friis) [reply src/utils/jsmnx.c:46]: done

**Review by torhaugl — APPROVED** [review]
The changes looks good. I approve. Merge when you want to

**Pull request merged**